lib/MqttClient.ts

```diff
@@ -391,7 +389,7 @@ class MqttClient extends EventEmitter {
 	_onMessageReceived(topic: string, payload: Buffer) {
 		if (this.closed) return
 
-		let parsed: any = payload ? payload.toString() : payload
+		let parsed: string | number | Record<string, any> = payload?.toString()
```

Comment:
`payload?.toString()` can be `undefined` if `payload` is undefined!

Reply:
added undefined to types